packages/medusa/src/services/totals.ts

```diff
@@ -1046,14 +1080,10 @@ class TotalsService extends TransactionBaseService {
       excludeNonDiscounts: true,
     })
 
-    if (!cartOrOrder.discounts || !cartOrOrder.discounts.length) {
-      return 0
-    }
-
     // we only support having free shipping and one other discount, so first
     // find the discount, which is not free shipping.
-    const discount = cartOrOrder.discounts.find(
-      ({ rule }) => rule.type !== "free_shipping"
+    const discount = cartOrOrder.discounts?.find(
```

Comment:
praise: Make sense to use it for the next constraint since if there is no discount we will not loop anyway 👍